src/coreclr/jit/hwintrinsic.cpp

```diff
@@ -1381,7 +1395,7 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
             }
 
 #if defined(TARGET_ARM64)
-            if ((simdSize != 8) && (simdSize != 16))
+            if ((simdSize != 8) && (simdSize != 16) && (simdSize != 0))
```

Comment:
Shouldn't we have a `simdSize` from the return type in the worst case here?

Reply:
That is, I feel like either `simdSize` should've been set or we should instead have something more like:
```csharp
if ((simdSize != 8) && (simdSize != 16))
{
    if ((simdSize != 0) || ((simdRetSize != 8) && (simdRetSize != 16))
    {
        // ...
    }
}
```

Something that explicitly checks the state we're expecting to be in here given the assumptions from the code above.